Update documentation for transition to quarterly releases (#4026)
projectcontour/contour: site/content/resources/support.md

This document describes which versions of Contour are supported by the Contour team.

## Supported releases

> Contour is changing both to quarterly releases and three supported releases.

Reviewer: ```suggestion
Contour is releases quarterly and supports three versions starting with Contour v1.20.
```

Reviewer 2: maybe remove `is` in the suggestion as well

> The first Contour version covered by the quarterly release cadence will be Contour v1.20, scheduled for late October 2021.

Reviewer: ```suggestion

```

Author: I don't understand why you want to remove this? Right now, we need to ensure that during this transition period, this is clearly explained for people who are using Contour. That's the point of all of this present-tense language. This will not live forever, once we are fully moved over to the new release schedule, then I will update this again.

As of 1.19, we are *not* releasing quarterly or supporting three versions. Saying otherwise is incorrect, and confusing.

> At the time it is released, it will be the only supported version, and versions 1.21 and 1.22 will continue supporting back to Contour 1.20.

Reviewer: ```suggestion

```

> When Contour 1.23 releases (nine months later), Contour 1.20 will fall out of support.

Reviewer: ```suggestion
When Contour 1.23 releases, Contour 1.20 will fall out of support.
```

> The following table illustrates how this will work. The given dates are estimates, not guarantees.

Author: If we don't say that the dates are estimates, then people can say "wait, you didn't release when you said you would".

> They are our best guess as to when each version will be released.

Reviewer: ```suggestion

```

| Version |v1.19 |v1.20|v1.21|v1.22|v1.23|
|---------|--------|-------|-------|-------|-------|
|Q3 2021 (September 2021) | :heavy_check_mark: |
|Q4 2021 (October 2021) | :negative_squared_cross_mark: | :heavy_check_mark: |

> |Q1 2022 (January 2021) | :negative_squared_cross_mark: | :heavy_check_mark: |:heavy_check_mark: |

Reviewer: ```suggestion
|Q1 2022 (January 2022) | :negative_squared_cross_mark: | :heavy_check_mark: |:heavy_check_mark: |
```

> |Q2 2022 (April 2021) | :negative_squared_cross_mark: | :heavy_check_mark: |:heavy_check_mark: |:heavy_check_mark: |

Reviewer: ```suggestion
|Q2 2022 (April 2022) | :negative_squared_cross_mark: | :heavy_check_mark: |:heavy_check_mark: |:heavy_check_mark: |
```

> |Q3 2022 (July 2021) | :negative_squared_cross_mark: | :negative_squared_cross_mark: |:heavy_check_mark: |:heavy_check_mark: | :heavy_check_mark: |

Reviewer: ```suggestion
|Q3 2022 (July 2022) | :negative_squared_cross_mark: | :negative_squared_cross_mark: |:heavy_check_mark: |:heavy_check_mark: | :heavy_check_mark: |
```

## What does a release being "supported" mean?

In short, "supported" means that Contour will issue fixes for security and other critical bugs for that release's supported lifetime.

However, the project will require users to upgrade to the most recent patch release for a version to be supported.

That is:
- The latest patch version in each release is the supported version.
- If you are not running the supported version from your release train, you'll need to upgrade first if you have any problems.
- When a new patch is cut, that will become the supported version for that release.

> So, if Contour 1.20.0 is the only supported version, and Contour 1.20.1 is released, then the supported version will change to 1.20.1.

Reviewer 3: wait, are you saying we will only support the latest patch of a supported minor?  So if 1.20 is a supported minor with 5 patches out, only 1.20.5 is supported?

Author: Yes. If not, then we have no way to fix problems. If there is a problem with 1.20.4, and we cut 1.20.5, how else do we say "to fix this problem, you have to upgrade"? This is pretty standard tbh.

> If, later on, the supported versions are 1.21, 1.22, and 1.23, and 1.21.1, 1.22.1, and 1.23.1 are released, then the patch releases will be the only supported versions.

Reviewer 3: so 1.21.0, 1.22.0, 1.23.0 will fall out of support?

Author: Yes. Is this not a standard feature of a patch release?

Reviewer 3: In thinking about this again, I don't think there's a difference btw the two so my question is moot I suppose.  The question I had in mind originally is if our stance is such that 1.20.5 comes out and renders anything <= 1.20.4 out of support, then what happens if another vulnerability is found on, say, 1.20.2. Presumably that same vulnerability still exists on the latest 1.20.5 because the delta btw 1.20.5 and the previous 1.20.4 is precisely some patched vulnerability.  So we'd be forced to release a 1.20.6.

Now if for some reason 1.20.5 was not prone to this vulnerability, I think we'd be fine recommending users to just upgrade to 1.20.5.

Author: Yes, that's how I expected everything to work too.

## Latest version and the `:latest` tag
The latest stable release is identified by the [Docker tag `:latest`][1].
`:latest` is an alias for {{< param latest_version >}} which is the current stable release.

> When required the project may release a patch release to address security issues, serious problems with no suitable workaround, or documentation issues.

Reviewer: What does this sentence mean, it's not clear to me what it's trying to tell the user. Just that we're going to, at any time, release a patch and that moves `:latest`?

Author: Yes. This is a *policy*. If we don't say exactly what `:latest` means, it's possible to misunderstand.

Reviewer: I guess the statement doesn't explain anything other than "we'll release something". I think what it's getting at is that the `:latest` tag will move at any time?
